$NetBSD: patch-config.h.in,v 1.1 2023/06/11 00:59:27 schmonz Exp $

Borrow upstream gnulib commit 0814a293:
Make static_assert work on Solaris 11.4.

--- config.h.in.orig	2022-10-16 20:23:50.000000000 +0000
+++ config.h.in
@@ -2298,4 +2298,10 @@
              && __GNUG__ < 6 && __clang_major__ < 6)))
  #include <assert.h>
  #undef/**/assert
+ /* Solaris 11.4 <assert.h> defines static_assert as a macro with 2 arguments.
+    We need it also to be invocable with a single argument.  */
+ #if defined __sun && (__STDC_VERSION__ - 0 >= 201112L) && !defined __cplusplus
+  #undef static_assert
+  #define static_assert _Static_assert
+ #endif
 #endif
